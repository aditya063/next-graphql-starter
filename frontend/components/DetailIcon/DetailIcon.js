import PropTypes from 'prop-types';
import Link from 'next/link';
import * as sc from './DetailIcon.style';

// https://github.com/zeit/next.js/issues/7915
const DetailIconRef = React.forwardRef((props, ref) => (
  <sc.DetailIcon
    {...props}
    forwardedRef={ref}
    type='up-square'
    theme='twoTone'
  />
));

DetailIconRef.displayName = 'DetailIconRef';

const DetailIcon = props => (
  <Link href={{ pathname: '/idea', query: { id: props.id } }} passHref>
    <DetailIconRef />
  </Link>
);

DetailIcon.propTypes = {
  id: PropTypes.string.isRequired
};

export default React.memo(DetailIcon);
