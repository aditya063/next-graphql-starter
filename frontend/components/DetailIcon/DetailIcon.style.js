import styled from 'styled-components';
import { Icon } from 'antd';

export const DetailIcon = styled(Icon).attrs({
  type: 'up-square',
  theme: 'twoTone'
})`
  font-size: 1.2rem;
`;
