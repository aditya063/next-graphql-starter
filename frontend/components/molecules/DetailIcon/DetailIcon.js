import PropTypes from 'prop-types';
import Link from 'next/link';
import { ExpandIconBtn } from '../../atoms/IconBtn/IconBtn';
// import { useRouter } from 'next/router';
// import * as sc from './DetailIcon.style';

const Icon = (props, ref) => (
  // const router = useRouter();

  <ExpandIconBtn
    className={props.className}
    aria-label='detail icon'
    // onClick={() =>
    //   router.push('/idea', `idea?id=${props.id}`, { id: props.id })
    // }
  />
);

const IconRef = React.forwardRef(Icon);

const DetailIcon = props => (
  // <Link href={`/idea/${props.id}`}>
  // as={`/idea/${props.id}`}
  <Link href={{ pathname: '/idea', query: { id: props.id } }}>
    <IconRef {...props} />
  </Link>
);

// http://localhost:8008/idea?id=ck5h7pmp1v1300b009uskbj2k

Icon.propTypes = {
  className: PropTypes.string
};

DetailIcon.propTypes = {
  id: PropTypes.string.isRequired
};

export default React.memo(DetailIcon);
