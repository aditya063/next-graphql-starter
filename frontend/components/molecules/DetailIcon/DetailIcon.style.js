import styled from 'styled-components';
import { Icon } from 'antd';

export const DetailIcon = styled(Icon)`
  font-size: 1.2rem;
`;

export const aLink = styled.a`
  line-height: 0;
`;
