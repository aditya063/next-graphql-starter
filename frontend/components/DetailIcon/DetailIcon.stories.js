import React from 'react';
import allCombos from '../../.storybook/allCombos';
import DetailIcon from './DetailIcon';

export default { title: 'DetailIcon', component: DetailIcon };

const props = {
  id: ['1']
};

export const all = () => allCombos(DetailIcon, props);
