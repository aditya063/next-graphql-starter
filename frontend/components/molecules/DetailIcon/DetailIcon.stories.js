import DetailIcon from './DetailIcon';

export default { title: 'Molecules', component: DetailIcon };

const data = {
  id: '1'
};

export const detailIcon = () => <DetailIcon {...data} />;
