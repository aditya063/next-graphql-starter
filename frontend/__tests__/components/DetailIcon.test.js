import { prettyDOM, render, fireEvent, wait } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import DetailIcon from '../../components/DetailIcon/DetailIcon';
import theme from '../../public/styles/theme.style';

describe('DetailIcon', () => {
  it('matches snapshot', () => {
    const mockProps = { id: '1' };

    const utils = render(
      <ThemeProvider theme={theme}>
        <DetailIcon {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
  });

  xit('has correct href', () => {
    const mockProps = { id: '1' };

    const utils = render(
      <ThemeProvider theme={theme}>
        <DetailIcon {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;
    console.log('TCL: container', prettyDOM(utils.baseElement));

    expect(container).toHaveAttribute('href', '/idea?id=1');
  });
});
