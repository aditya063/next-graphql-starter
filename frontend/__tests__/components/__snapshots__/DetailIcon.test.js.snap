// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailIcon matches snapshot 1`] = `
.c1 {
  font-size: 1.2rem;
}

.c0 {
  line-height: 0;
}

<a
  class="c0"
>
  <i
    aria-label="icon: up-square"
    class="anticon anticon-up-square c1"
  >
    <svg
      aria-hidden="true"
      class=""
      data-icon="up-square"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
        fill="#1890ff"
      />
      <path
        d="M184 840h656V184H184v656zm143.5-228.7l178-246c3.2-4.4 9.7-4.4 12.9 0l178 246c3.9 5.3.1 12.7-6.4 12.7h-46.9c-10.2 0-19.9-4.9-25.9-13.2L512 465.4 406.8 610.8c-6 8.3-15.6 13.2-25.9 13.2H334c-6.5 0-10.3-7.4-6.5-12.7z"
        fill="#e6f7ff"
      />
      <path
        d="M334 624h46.9c10.3 0 19.9-4.9 25.9-13.2L512 465.4l105.2 145.4c6 8.3 15.7 13.2 25.9 13.2H690c6.5 0 10.3-7.4 6.4-12.7l-178-246a7.95 7.95 0 0 0-12.9 0l-178 246c-3.8 5.3 0 12.7 6.5 12.7z"
        fill="#1890ff"
      />
    </svg>
  </i>
</a>
`;
